### ISSUE TITLE
precipitation isn't bias-correcting properly

### ISSUE THREAD

**Issue by cmahony** [bcgov/climr#245]
The 1961-1990 normals for models and observations don't match up for precipitation. since this is the reference period, everything should have the same 1961-1990 mean. something is wrong. 

```
my_points <- data.frame(lon = c(-127.7300,-127.7500), lat = c(55.34114, 55.25), elev = c(711, 500), id = 1:2)
my_data <- plot_timeSeries_input(my_points) # generate the input data
plot_timeSeries(my_data, "MAP")
```
![image](https://github.com/bcgov/climr/assets/17104005/ef03c324-d80a-4364-bedf-e23160fbd588)

**Comment by kdaust** [comment]
@cmahony yes I know there are some issues with the climatena historic ts - hopefully the updated version (tomorrow) should fix the issue. Could you check if precip looks on these plots with the cru_gpcc now it's fixed?

**Comment by cmahony** [comment]
the CRU time series is fixed for temperature, but has the same problem for precipitation. I have a hunch what may be the problem. we should check if this is a problem wiht monthly precip (i can't do that in the plot function becuase we need to change the variable naming conventions first... something i wanted to check in with you on). 
![image](https://github.com/bcgov/climr/assets/17104005/8a7b7835-3bbe-4fdf-bc04-420e0f0d5622)

**Comment by cmahony** [comment]
@kdaust i did a basic check of the data and it looks like this is a problem with my function, not with the database or downscaling. 

```
library(data.table)
library(scales)
library(stinepack)

# data frame of arbitrary points
my_points <- data.frame(lon = c(-127.7300), lat = c(55.34114), elev = c(711), id = 1)

# generate the input data
my_data <- plot_timeSeries_input(my_points, historic_ts_dataset = "cru_gpcc", gcm_models = list_gcm()[1])

# use the input to create a plot
plot_timeSeries(my_data, "MAP", historic_ts_dataset = "cru_gpcc", gcm_models = list_gcm()[1])

variable <- "MAP"
baseline <- my_data[PERIOD=="1961_1990", get(variable)]
x.obs <- as.numeric(my_data[PERIOD%in%1900:2100, "PERIOD"][[1]])
y.obs <- my_data[PERIOD%in%1900:2100, get(variable)]
baseline
mean(y.obs[x.obs%in%1961:1990])
```

```
> baseline
[1] 619.2334
> mean(y.obs[x.obs%in%1961:1990])
[1] 618.1211
```

**Comment by cmahony** [comment]
it was a problem in my function. the bug is fixed in 0317513cd8feb303ccd27cb1bde0b4a2621ce6d4
